test/TestProviderRound.spec.js

```diff
@@ -98,6 +98,41 @@ contract('ProviderRound', accounts => {
 
   });
 
+  describe('resignAsProvider', () => {
+
+    before(async () => {
+      providerRound = await ProviderRound.new();
+      await blockMiner.mineUntilEndOfElectionPeriod(providerRound);
+      await providerRound.initializeRound();
+    });
+
+    it('should remove a provider from the provider mapping', async () => {
+      await providerRound.provider(22, 10, 1, 25, {from: accounts[0]});
+      const registeredProvider = await providerRound.providers.call(accounts[0]);
+      assert.equal(PROVIDER_REGISTERED, registeredProvider[0])
+      await providerRound.resignAsProvider({from: accounts[0]});
+      const resignedProvider = await providerRound.providers.call(accounts[0]);
+      for(param of resignedProvider) {
```

Comment:
This would fail if the provider struct had any other kind of type. I would explicitly check each field, here.